crd-generator/maven-plugin/README.md

```diff
@@ -0,0 +1,180 @@
+# CRD-Generator - Maven Plugin
+
+Maven plugin for the CRD-Generator: Generate CRDs from Java model during Maven builds.
```

Comment:
This document should be either linked in https://github.com/fabric8io/kubernetes-client/blob/main/doc/CRD-generator.md or we should move it in `doc/` folder like the rest of the documentation files.

Reply:
Good point, we havn't talked about where the documentation for the generator-v2 should be placed, how it should be structured and especially how it should be distinguished to generator-v1.

My intention was to write first for each generator-v2 extension (maven-plugin, gradle-plugin, cli) an own README until at least two of them are merged. 
Afterwards, it should be easier to write some common notes about generator-v2 including migration paths and deprecations / roadmap into the central docs. And of course the content from the README's can be moved again, too.

If I should do this within this PR, I need some guidence on how to introduce generator-v2. Otherwise we should probably create a dedicated issue for the common part of the generator-v2 docs.

Reply:
@rohanKanojia @manusa 

Are you ok with this strategy or do we need more in this MR? (Can we resolve this conversion?)

Reply:
I'm okay with this strategy. Let's have something up and running first then we can interate	over how to better organize documentation.